### DISCUSSION TITLE
Showing WinForms View from Avalonia Project

### DISCUSSION THREAD

**Discussion by vector-man** [AvaloniaUI/Avalonia#8505] [Q&A]
Does anyone know if it's possible to show a WinForms dialog from an Avalonia View? I'm migrating and am working on the main view, while keeping the WinForms views for now, until I get time to redesign them.

The Avalonia project is separate, but I want to reference the WinForms project and show the dialogs over the Avalonia one, basically.

I've coded everything in ReactiveUI from the beginning, including the WinForms project.

**Comment by maxkatz6** [comment]
If you want to embed winforms control inside of the avalonia window, you can do it with NativeControlHost. 
If sometimes you need the opposite, to embed Avalonia control inside of WinForms window, you would need to use [WinFormsAvaloniaControlHost](https://github.com/AvaloniaUI/Avalonia/tree/master/samples/interop/WindowsInteropTest).

But if you just need to open a WinForms window side by side with Avalonia windows, there shouldn't be any problem at all. Form.Show() should just work, if winforms code was initialized before in Program.cs.
